OpenRA.Mods.Common/Activities/FindAndDeliverResources.cs

```diff
@@ -122,10 +122,10 @@ public override bool Tick(Actor self)
 			// of the refinery entrance.
 			if (LastSearchFailed)
 			{
-				var lastproc = harv.DockClientManager?.LastReservedHost;
+				var lastproc = harv.LinkClientManager?.LastReservedHost;
 				if (lastproc != null)
 				{
-					var deliveryLoc = self.World.Map.CellContaining(lastproc.DockPosition);
+					var deliveryLoc = self.World.Map.CellContaining(lastproc.LinkPosition);
```

Comment:
This is a good example of my comment above: does a `LinkPosition` even make sense for a proximity-style link? This needs to become an implementation detail of the dock style linkage, and then the proximity style linkage would need its own version that considers all suitable cells within range.

Reply:
it makes sense in that for proximity docking this is the centre of the proximity dock

Reply:
The problem is that its at best misleading, and at worst likely to lead to completely wrong behaviour.

Consider a LinkHost that's on the wrong side of an impassable cliff boundary. The host exposes a LinkPosition property, so its completely reasonable to expect that we can/should evaluate a pathfind operation to that position. Since that position isn't pathable, the client won't attempt to Link. This is the correct and expected behaviour for a DockHost. For a ProximityLinkHost it isn't - there are pathable cells within range that the client should be considering instead.

Reply:
the ProximityLinkHost queues a MoveWithinRange activity so it does consider other positions

Reply:
That doesn't address the issue with the abstraction.

Reply:
I'm not completely happy with the current `ILinkHost` abstraction as well. It restrains to just one possible facing, and a few other things. I think it's a bit out of scope in this PR